portal-impl/src/com/liferay/portlet/documentlibrary/util/DLPermissionPropagationUtil.java

```diff
@@ -18,6 +19,30 @@
  */
 public class DLPermissionPropagationUtil {
 
+	public static long getInheritableParentFolderId(DLFileEntry dlFileEntry)
+		throws PortalException {
+
+		DLFolder dlFolder = dlFileEntry.getFolder();
```

Comment:
check in here, for HomeFolder, dlFileEntry is null

Reply:
I already checked, it's not null

Reply:
It seems we have to check the Propagation permission twice in case the folder is the Home folder.
I think it would be better to create a list of parentFolderId excluded Home folderId and then calling the function
```
	public static long getInheritableParentFolderId(
		long companyId, long groupId, List<Long> ancestorIds)
```